modules/viewHierarchyMixin.js

```diff
@@ -106,49 +111,201 @@
      * @method cleanupSelf
      */
     cleanupSelf: function() {
+      this.deactivate();
+
       // Clean up child views first
       this.cleanupChildViews();
 
+      // Remove view from DOM
+      this.remove();
+
       // Unbind all local event bindings
       this.unbind();
       this.off();
       this.stopListening();
 
-      // Remove view from DOM
-      this.remove();
-
-      // Undelegates events
-      this.undelegateEvents();
-
       // Delete the dom references
       delete this.$el;
       delete this.el;
     },
 
     /**
+     * Method to be invoked when deactivate is called. Use this method to turn off any
```

Comment:
mention that this is a no-op by default (see Backbone documentation http://backbonejs.org/#Model-parse for an example of explicitly saying its a no-op).